### ISSUE TITLE
Eliminate stutter when camera and synthesizer are running concurrently

### ISSUE THREAD

**Issue by tonystuart** [tonystuart/jamming#1]
There is a distracting stutter at the beginning of each note when the camera and the synthesizer are running concurrently. For a single object (e.g. a 1 inch by 1 inch post it note), there are at least three echoes as the note fades. Here are some things I noticed:
- The stutter stops immediately when the camera process is terminated
- The stutter does not occur when the camera is not running, even if the image is updated
- The image update can consist of simply updating the time stamp or copying another image
- The stutter occurs even when the camera process is running in another directory

Based on these observations, it seems that the problem is due to the interaction between the camera and the synthesizer, but not due to the interaction between the image and the image processor.

**Comment by tonystuart** [comment]
Installed fluidsynth to see if its ability to lock pages into memory and run at a real time priority would solve the problem. Unfortunately the problem still exists. Running just the image capture program at the same time that fluidsynth is playing a MIDI file causes repeated and persistent audio drop outs.

Using top to get a rough idea of CPU utilization, raspistill is only taking about 0.3% of one CPU, while fluidsynth is taking 50-80% of one CPU with 188m of memory resident and about 80m free.

Note that fluidsynth by itself has trouble playing the test MIDI file on the Pi model 1B, but plays it fine on the model 2. However, it has problems playing more complex MIDI files even on the model 2.

**Comment by tonystuart** [comment]
Here is the raspistill command line:

raspistill -rot 270 -w 1024 -h 768 -br 30 -tl 1000 -t 21600000 -n -o Jamming.tmp -l Jamming.jpg

**Comment by tonystuart** [comment]
Changing the time-lapse parameter to a large number eliminates the stutter, so it appears to be the act of capturing the image that causes the problem. A solution may be to capture the image between the end and beginning of the loop.

**Comment by tonystuart** [comment]
I modified the approach so that the image capture and analysis functions are now performed sequentially instead of in parallel. This ensures that the image capture does not occur while the synthesizer is running and prevents the stutter.

Although this approach results in an increase in latency, the user experience may actually be better because there is now a clear expectation that any changes made within the camera's field of view will be applied before the start of the next loop. In addition the implementation is more straightforward.

The new approach wraps the raspistill command in a Java class to make it easier to start and stop raspistill and to take advantage of the raspistill keypress command line option for triggering the capture. In addition it uses a WatchService to detect the linked file that raspistill creates when the image is captured.

As part of this activity, I added new command line options for raspistill and rationalized the existing options.
